src/GenSymIO.chpl

```diff
@@ -1908,17 +1929,17 @@ module GenSymIO {
                 firstSeg = 0;
                 lastSeg = segsArray.front();
```

Comment:
I know this line wasn't changed in the PR, but I have a question. about the `lastSeg` in the first locale.  Isn't the first "segment" (aka offset) in the `segsArray` going to be 0 in the first locale?  i.e. The first string starts at offset 0.  So segsArray.front() on local-0 should have the value of 0?  NOTE: When I construct a sample Strings array and inspect the `Strings.offsets` in the client, it shows the first element being zero.
```
>>> s = ak.array(["abc", "def", "xyz"])
>>> str(s.offsets)
'[0 4 8]'
```

If the goal is to find the end of the first segment then shouldn't it be segsArray[1] in the first local?

Moreover, isn't this the wrong goal anyway?  Isn't the real thing we want to find "the start position of the last segment" which would be `lastSeg = segsArray.back()`, the same as in the else clause?

Reply:
Ah, yeah, will take a closer look at this later today

Reply:
@glitch this is an excellent catch as you are correct in your thoughts. This bug was never caught because the biz rules are such that the majority of shuffles are shuffle lefts as shown in this example:

![image](https://user-images.githubusercontent.com/10785153/118357890-4e40a980-b54a-11eb-832d-166f5fc315c0.png)

Accordingly, chars are shuffled left from locale 1 to locale 0 most of the time and therefore the segments and values are spread evenly amongst the locales. An example is below, with the current logic in the master and feature branches listed first, and the corrected version listed second:

![image](https://user-images.githubusercontent.com/10785153/118356460-4fbaa380-b543-11eb-9a21-f5e2bb06a603.png)

![image](https://user-images.githubusercontent.com/10785153/118357589-e2aa0c80-b548-11eb-9714-71e9681398de.png)

Again, this is an excellent catch -> thank you!

Reply:
And here's the corrected logic with logging statements:

![image](https://user-images.githubusercontent.com/10785153/118362548-eb590d80-b55d-11eb-83a7-d9afe546552b.png)

![image](https://user-images.githubusercontent.com/10785153/118362510-ca90b800-b55d-11eb-8e4a-f770e9f9ab56.png)

Reply:
[Committed](https://github.com/hokiegeek2/arkouda/commit/bbd874e4a5e4532b5e222df2e779f6f3abaa65b2) corresponding code change after extensive testing, all set.